Create authorization doc (#614)
kyma-project/test-infra: docs/prow/authorization.md

# Authorization

## Required GCP Permissions

To deploy a Prow cluster, configure the following service accounts in the GCP project you own.

| Service account name          | Usage                                                      | Required roles |
| :---------------------------- | :----------------------------------------------------------| :------------- |
| **sa-gcs-plank**              | Used by Prow plan microservice | `Storage Object Admin`
| **sa-gke-kyma-integration**   | Running integration tests on GKE cluster | `Compute Admin`, `Kubernetes Engine Admin`, `Kubernetes Engine Cluster Admin`, `DNS Administrator`, `Service Account User`, `Storage Admin`
| **sa-kyma-artifacts**         | Saving release artifacts to the GCS bucket | `Storage Object Admin`
| **sa-vm-kyma-integration**    | Running integration tests on minikube | `Compute Instance Admin (beta)`, `Compute OS Admin Login`, `Service Account User`
| **sa-gcr-push-kyma-project**  | Publishing docker images | `Storage Admin`

> ## Kubernetes RBAC rules on Prow cluster

Reviewer: Done. Section fixed.

| **tide** | - `prowjobs.prow.k8s.io` |  create, list  |

## User permissions on GitHub

Prow is responsible for starting tests in reaction to certain Github events. For security reasons, the `trigger` plugin ensures that test jobs are run only on pull requests created or verified by trusted users.

### Trusted users
Members of the `kyma-project` organization are considered trusted users. Trigger starts jobs automatically when a trusted user opens a pull request or commits changes to a pull request branch. Alternatively, trusted collaborators can start jobs manually via the `/test all`, `/test {JOB_NAME}` and `/retest` commands, even if a particular pull request was created by an external user. 

### External contributors
External contributors are users outside the `kyma-project` organization. Trigger does not automatically start test jobs on pull requests created by external contributors. Furthermore, external contributors are not allowed to manually run tests on their own pull requests.

> **NOTE:** External contributors can still trigger tests on pull requests created by trusted users.

## Authorization decisions enforced by Prow

> Action on Prow can be only triggered by webhooks. To configure them you need to provide two secrets:

Reviewer 2: ```suggestion
Actions on Prow can be triggered only by webhooks. To configure them you must provide two secrets:
```

Reviewer 2: Where do you "provide the secrets"? SECRETS as in "Kubernetes <Secret> object"?

> - hmac-token - used to validate webhook

Reviewer 2: ```suggestion
- `hmac-token` - used to validate webhook
```

> - oauth-token - GitHub bot access token

Reviewer 2: ```suggestion
- `oauth-token` - stores the access token used by the GitHub bot
```

### Trusted users
All members of the `kyma-project` organization are considered trusted users. The `trigger` plugin starts jobs automatically when a trusted user opens a PR or commits changes to a PR branch. Alternatively, trusted collaborators can start jobs manually through the `/test all`, `/test {JOB_NAME}` and `/retest` commands, even if a particular pull request was created by an external user. 

### External contributors
All users that are not members of the `kyma-project` organization are considered external contributors. The `trigger` plugin does not automatically start test jobs on PRs created by external contributors. Furthermore, external contributors are not allowed to manually run tests on their own pull requests.

> **NOTE:** External contributors can still trigger tests on pull requests created by trusted users.

## Authorization decisions enforced by Prow

Actions on Prow can be triggered only by webhooks. To configure them you must create two Github Secrets on your Prow cluster:
- `hmac-token` - used to validate webhook
- `oauth-token` - stores the access token used by the GitHub bot

> Follow the official [Prow documentation](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#create-the-github-secrets) to learn how create the secrets.

Reviewer 2: ```suggestion
Follow the official [Prow documentation](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#create-the-github-secrets) to learn how to create the Secrets.
```
